lib/rules/identical-links-same-purpose-matches.js

```diff
@@ -0,0 +1,79 @@
+const { aria, text, dom } = axe.commons;
+const { escapeSelector } = axe.utils;
+
+/**
+ * Note:
+ * Firefox does not respect map > area as visible,
+ * even when used by a img[usemap], hence the usage of `includeHidden` flag,
+ * in the accessible name computation
+ */
+const hasAccName = !!text.accessibleTextVirtual(virtualNode, {
+	includeHidden: true
+});
+if (!hasAccName) {
+	return false;
+}
+
+const role = aria.getRole(node);
+if (role && role !== 'link') {
+	return false;
+}
+
+const nodeName = node.nodeName.toUpperCase();
+if (['A', 'AREA'].includes(nodeName) && !node.hasAttribute('href')) {
+	return false;
+}
+
+if (nodeName === 'AREA') {
+	const rootEl = dom.getRootNode(node);
+	/**
+	 * Ensure map is used by `img` witin the document tree
+	 */
+	const mapEl = getMapForAreaElement(node, rootEl);
```

Comment:
I have my doubts that you can put `area` elements in a different tree from the `map` element. Please test that. In which case `node.closest('map')` should do fine. If not, use `dom.findUpVirtual(virtualNode, 'map')`. This new `getMapForAreaElement`  seems unnecessary.